src/scripts/generate_binary_ops.py

```diff
@@ -138,24 +138,27 @@ def sub(self, left_value, right_value):
 
     # Multiplication
     def mul(self, left_value, right_value):
-        result = left_value * right_value
-        if result < _constants.min: # Minimum / maximum number to safely multiply operand 1
-            right_value = _constants.min / left_value
-        elif result > _constants.max:
-            right_value = _constants.max / left_value
-        right_value = int(right_value) # Rounding because it might not be an integer
+        if left_value != 0:
+            if left_value < 0:
+                start = math.ceil(_constants.max / float(left_value))
+                end = math.floor(_constants.min / float(left_value))
+                right_value = random.randint(start, end)
+            if left_value > 0:
+                start = math.ceil(_constants.min / float(left_value))
+                end = math.floor(_constants.max / float(left_value))
+                right_value = random.randint(start, end)
+
         result = left_value * right_value
         return self.create_operation_result_object('*', right_value, result)
 
     # Division (floor division)
     def div(self, left_value, right_value):
-        if right_value == 0: # To avoid division by zero
-            if (random.randint(0, 1)):
-                right_value = random.randint(1, _constants.max)
-            else:
-                right_value = random.randint(_constants.min, - 1)
+        right_value = random.randint(_constants.min, _constants.max)
+        if right_value == 0:
+            right_value = random.randint(1, _constants.max)
         result = left_value / right_value
-        result = int(result) # Rounding because it might not be an integer
```

Comment:
If you are about to fix 2.7 please check both 2.7 and 3.x as well. The current version generates different values!
Consider the differences of the division operator `/` between 2.7 and 3.x!

Lets see if the random returns 9999
3.x:
In [39]: div(-9998,0)
Out[39]: ['/', 9999, 0.00010001000100012813]
In [40]: div(-9999,0)
Out[40]: ['/', 9999, 0.0]
In [41]: div(-10000,0)
Out[41]: ['/', 9999, -0.00010001000100001711]

2.7:
In [5]: div(-9998,0)
Out[5]: ['/', 9999, 0]
In [6]: div(-9999,0)
Out[6]: ['/', 9999, 0]
In [7]: div(-10000,0)
Out[7]: ['/', 9999, -1]

Reply:
I fixed it. Both 2.7 and 3.6 generates the right value now